enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,342 @@
+   cluster.
+
+## Proposal
+
+### User Stories
+
+#### Add RHEL content to running containers
+
+As a developer or application administrator
+I want access to RHEL subscription content in my containers
+So that I can install RHEL content within my container via `yum install`
+
+#### Access RHEL content from a Satellite instance
+
+As a developer using OpenShift 
+I want to be able to access RHEL content from my Satellite instance
+So that I download RHEL content from my Satellite instances instead of the Red Hat CDN.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Installation and Setup
+
+When the `subscription-injection-operator` is installed, it creates a subscription `ClusterBundle`
+custom resource definition. The operator deployment creates a default instance which is
+cluster-scoped:
+
+```yaml
+kind: ClusterBundle
+apiVersion: subscription.openshift.io/v1
+metadata:
+  name: cluster
+spec:
+  entitlement:
+    name: etc-pki-entitlement
+```
+
+The `entitlement` object is a reference to a `Secret` in the `openshift-config` namespace
+containing the entitlement keys for the cluster's subscription. The cluster administrator is
+responsible for obtaining the `entitlement` keys and adding them to the cluster. Administrators can
+refer to existing guidance on how to obtain this information.
+
+#### Subscription Bundle Controller
+
+When a `ClusterBundle` object is created, the Subscription Bundle Controller (a new controller)
+will perform the necessary actions to make the bundle available to cluster workloads. This consists
+of the following:
+
+1. Verify that the referenced `entitlement` and optional `yumRepository` resources exist in the
+   `openshift-config` namespace.
+2. Create the ProjectedResource CSI driver instances for the `entitlement` and (optional)
+  `yumRepository` resources as declared in the Projected Resource CSI Driver proposal.
+3. Create a `ClusterRole` associated with the bundle, which grants `get`, `watch`, and `list`
+   permissions to the associated shares and the `ClusterBundle` object.
+4. Aggregate the cluster role to a configurable list of system roles. By default this will be
+   aggregated to the `edit` and `admin` roles.
+5. Add owner references to the generated `ClusterRole` and shares. This ensures the dependent
+   objects are deleted when the `ClusterBundle` is deleted.
+
+#### Subscription Injection
+
+Developers can inject subscription bundles by adding the
+`subscription.openshift.io/inject-cluster-bundle` annotation to a pod template spec. Any workload
+controller which ultimately creates a `Pod` will be supported. For example, to inject a
+subscription into a `Deployment`, use the following YAML:
+
+```yaml
+kind: Deployment
+apiVersion: v1
+metadata:
+  name: my-rhel-deployment
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: "rhel-deployment"
+  template:
+    metadata:
+      annotations:
+        "subscription.openshift.io/inject-cluster-bundle": cluster
```

Comment:
implication here is that a given pod can only have a single clusterbundle injected, right?

Reply:
In my experience, it's common to have multiple certificates, so I'd rather see them supported than not.

Reply:
So I think this can be addressed by either:

1. Instructing users to add all desired keys into one `Secret` if they intend on using those certificates together (say, in a layered product)
2. Allow multiple `Secret` objects per bundle.

Option 2 is feasible IMO and does not preclude 1.

Reply:
all in one secret is less ideal since it may be more work for the admin, if they need to create various disjoint sets for different users.  But certainly it's "sufficient".  I'm less worried about having the functionality on day one, but the bigger problem is if we pidgeonhole ourselves into a single bundle api and then have a hard time extending it to mulitple in the future.

Reply:
@bparees Updated to allow multiple bundles to be injected.